model/src/clients/testsys_client.rs

```diff
@@ -0,0 +1,1301 @@
+            Self::Resource(resource) => resource.metadata.name.to_owned(),
+        }
+    }
+
+    fn create_result_row(&self) -> Vec<ResultRow> {
+        let mut results = Vec::new();
+        match self {
+            Crd::Test(test) => {
+                let name = test.metadata.name.clone().unwrap_or_else(|| "".to_string());
+                let state = test.test_user_state().to_string();
+                let test_results = &test.agent_status().results;
+                if test_results.is_empty() {
+                    results.push(ResultRow {
+                        name,
+                        object_type: "Test".to_string(),
+                        state,
+                        passed: None,
+                        skipped: None,
+                        failed: None,
+                    })
+                } else {
+                    for (test_count, result) in test_results.iter().enumerate() {
+                        let retry_name = if test_count == 0 {
+                            name.clone()
+                        } else {
+                            format!("{}-retry-{}", name, test_count)
+                        };
+                        results.push(ResultRow {
+                            name: retry_name,
+                            object_type: "Test".to_string(),
+                            state: state.clone(),
+                            passed: Some(result.num_passed),
+                            skipped: Some(result.num_skipped),
+                            failed: Some(result.num_failed),
+                        });
+                    }
+                }
+            }
+            Crd::Resource(resource) => {
+                let name = resource.name();
+                let mut create_state = TaskState::Unknown;
+                let mut delete_state = TaskState::Unknown;
+                if let Some(status) = resource.status() {
+                    create_state = status.creation.task_state;
+                    delete_state = status.destruction.task_state;
+                }
+                let state = match delete_state {
+                    TaskState::Unknown => create_state,
+                    _ => delete_state,
+                };
+
+                results.push(ResultRow {
+                    name,
+                    object_type: "Resource".to_string(),
+                    state: state.to_string(),
+                    passed: None,
+                    skipped: None,
+                    failed: None,
+                });
+            }
+        };
+        results
+    }
+}
+
+impl From<Crd> for CrdName {
+    fn from(crd: Crd) -> Self {
+        match crd {
+            Crd::Test(test) => CrdName::Test(test.name()),
+            Crd::Resource(resource) => CrdName::Resource(resource.name()),
+        }
+    }
+}
+
+#[derive(Debug)]
+pub enum DeleteEvent {
+    Starting(CrdName),
+    Deleted(CrdName),
+    Failed(CrdName),
+}
+
+pub enum SelectionParams {
```

Comment:
Possibly `SelectionParams` and `Status` and `ResultsRow` could go in their own file.

I'm not actually sure that all of this belongs in the `clients` module. This collection of files might be its own module directory as a sibling to `clients`.